hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/crypto/key/TestKeyProvider.java

```diff
@@ -62,6 +62,12 @@ public void testParseVersionName() throws Exception {
     } catch (IOException e) {
       assertTrue(true);
     }
+    try {
```

Comment:
LambdaTestUtils.intercept, which is 
1. simpler
2. returns the value of getBaseName in the exception
3. rethrows the caught exception if class/name doesn't match

Reply:
Thank you @steveloughran for the review, 
we have used LambdaTestUtils.intercept instead of try.